tree/treeplayer/inc/ROOT/TDFUtils.hxx

```diff
@@ -35,6 +35,16 @@ namespace ROOT {
 namespace Experimental {
 template <int D, typename P, template <int, typename, template <typename> class> class... S>
 class THist;
+
+namespace TDF {
+struct SnapshotOptions {
+  Int_t autosave   = 0;
+  Int_t compress   = 1;
```

Comment:
The data members here need documentation since users will use them.
Commonly in TDF we used `int` rather than `Int_t` but it does not make that much of a difference :)

Reply:
P.S.
maybe @pcanal or @Axel-Naumann know if there is a way to retrieve the default values without hardcoding them?

Reply:
The typical way is to add them to config/rootrc.in (which becomes $ROOTSYS/etc/system.rootrc) and retrieve the default via 'gEnv'.

Reply:
I will add proper documentation in the next push. Unfortunately, I don't think that we can figure out what the default parameter to other functions are. About `int` vs `Int_t`, if we use the former, we're asking for trouble if it becomes inconsistent with other parts of ROOT (i.e., in the future `int` may be 64bits wide...). I'm using `Int_t` because the functions I'm passing the value to take `Int_t`.

Reply:
@pcanal it would be nice to be able to recover the defaults that TTree and TFile use for these parameters, but I guess they are not in `gEnv`

Reply:
Indeed, they are not.  they are not customizable either and they are not available via any constexpr.  It might be sense (in a separate PR) to add either a public constexpr with the default value or a set of interface returning the default (and changeable via rootrc).